tests/models/test_hooks.py

```diff
@@ -234,11 +234,55 @@ class HookedModel(BoringModel):
     def __init__(self):
         super().__init__()
         self.called = []
+        self.train_batch = [
+            'on_train_batch_start',
+            'on_before_batch_transfer',
+            'transfer_batch_to_device',
+            'on_after_batch_transfer',
+            'training_step',
+            'on_before_zero_grad',
+            'optimizer_zero_grad',
+            'backward',
+            'on_after_backward',
+            'optimizer_step',
+            'on_train_batch_end',
+        ]
+        self.val_batch = [
+            'on_validation_batch_start',
+            'on_before_batch_transfer',
+            'transfer_batch_to_device',
+            'on_after_batch_transfer',
+            'on_validation_batch_end',
+        ]
+
+    def training_step(self, *args, **kwargs):
```

Comment:
wahy not use Mock? @awaelchli

Reply:
It's hard to use as there are several checks throughout our codebase for `isinstance(LightningModule)`

Reply:
ok, so lest check if we have a ticket for mocking this kind of tests and if note create one...

Reply:
I don't think it's possible, maybe if we could mock the inherited classes defining all these hooks...